Sources/CombineSchedulers/Timer.swift

```diff
@@ -232,8 +232,8 @@
         }
       }
 
-      public func receive<S: Subscriber>(subscriber: S)
-      where Failure == S.Failure, Output == S.Input {
+      public func receive<SC: Subscriber>(subscriber: SC)
+      where Failure == SC.Failure, Output == SC.Input {
```

Comment:
This `SC` generic is enough of an outlier that I think it'll probably get undone. I think we should take a few steps before we merge this (we can take the work on if you'd like):

1. We should get library evolution into CI to avoid accidentally renaming this back to `S`.
2. We should rename `SC` to `Subscriber`. Shadowing is more of our "style":

    ```suggestion
          public func receive<Subscriber: Combine.Subscriber>(subscriber: Subscriber)
          where Failure == Subscriber, Output == Subscriber.Input {
    ```

3. We should file a bug with Apple (or find the bug) and note it here in an "`NB`" comment.

Reply:
Sure. If you plan to rework this, you can definitely just discard the PR.
I can file a bug with Apple, although it seems that this is a know problem that Apple chooses to ignore (unfortunately).

There are similar problems in the TCA repo/code in the prerelease branch (different Apple bugs, probably), so getting library evolution test in the CI would surely help.

Reply:
> There are similar problems in the TCA repo/code in the prerelease branch (different Apple bugs, probably), so getting library evolution test in the CI would surely help.

Could you explain a bit more about the bugs in TCA? We have CI running on the `prerelease/1.0` and it does build for library evolution, and so far it is green.

Does the problem surface when building for library evolution, or is it just when trying to use an xcframework that the problem surfaces?

Reply:
Indeed, it's only when library evolution is used and .swiftinterface files are generated.
The problems are seen NOT when you compile TCA itself, but when some other app tries to import the xcframework (i.e. when the Swift compiler tries to compile the .swiftinterface files)

Here is an example problem I get.

In the .swiftinterface file the compiler generates the following for the enum `_Conditional`, pulling some things from the ReducerProtocol, namely the _Body type definitions (look at the last line in the snippet below).

```swift
public enum _Conditional<First, Second> : ComposableArchitecture.Reducer where First : ComposableArchitecture.Reducer, Second : ComposableArchitecture.Reducer, First.Action == Second.Action, First.State == Second.State {
    case first(First)
    case second(Second)
    @inlinable public func reduce(into state: inout First.State, action: First.Action) -> ComposableArchitecture.Effect<First.Action> {
      switch self {
      case let .first(first):
        return first.reduce(into: &state, action: action)

      case let .second(second):
        return second.reduce(into: &state, action: action)
      }
    }
    public typealias Action = First.Action
    public typealias State = First.State
    public typealias _Body = ComposableArchitecture.ReducerBuilder<State, Action>._Conditional<First, Second>.Body
  }
```

The error that I get from Swift is:

```
Type alias '_Body' references itself.
```

The relevant lines from the ReducerProtocol from which the line `public typealias _Body = ComposableArchitecture.ReducerBuilder<State, Action>._Conditional<First, Second>.Body` seemingly gets generated are these:

```swift

    associatedtype _Body

    typealias Body = _Body

```